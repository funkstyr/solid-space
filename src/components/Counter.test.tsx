import { fireEvent, render, screen } from '@solidjs/testing-library';
import Counter from './Counter';

describe('<Counter />', () => {
  it('it starts with zero', () => {
    render(() => <Counter />);

    const button = screen.getByRole('button');

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent('Clicks: 0');
  });

  it('it increases its value on click', async () => {
    render(() => <Counter />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    // the event loop takes one Promise to resolve to be finished
    await Promise.resolve();

    expect(button).toHaveTextContent('Clicks: 1');

    fireEvent.click(button);
    await Promise.resolve();

    expect(button).toHaveTextContent('Clicks: 2');
  });

  it('increments value', async () => {
    const { queryByRole, unmount } = render(() => <Counter />);

    const button = (await queryByRole('button')) as HTMLButtonElement;

    expect(button).toBeInTheDocument();
    expect(button).toHaveTextContent(/Clicks: 0/);

    fireEvent.click(button);

    expect(button).toHaveTextContent(/Clicks: 1/);
    unmount();
  });

  it('renders 1', () => {
    const { container, unmount } = render(() => <Counter />);
    expect(container).toMatchSnapshot();
    unmount();
  });
});
