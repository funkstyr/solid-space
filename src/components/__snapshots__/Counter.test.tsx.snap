// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Counter /> > renders 1 1`] = `
<div>
  <button
    class="w-[200px] rounded-full bg-gray-100 border-2 border-gray-300 focus:border-gray-400 active:border-gray-400 px-[2rem] py-[1rem]"
  >
    Clicks: 
    <!--#-->
    0
    <!--/-->
  </button>
</div>
`;
